### PR TITLE
kie-issues#36: Static validation for BPMN files on KIE Sandbox

### LINKED ISSUE #36: Static validation for BPMN files on KIE Sandbox
https://github.com/kiegroup/kie-issues/issues/14 provide implementation of server-side BPMN validation (jitexecutor).

- Make the application hosting the BPMN Editor consume this new endpoint.
- Make sure that the output appears on the Problems tab

Rest enpoint accept the following json:

```

{
  "mainURI": "string",
  "resources": [
    {
      "URI": "string",
      "content": "string"
    }
  ]
}
```

For each model, a 

```
 {
      "URI": "string",
      "content": "string"
    }
```

must be provided

Ind the UI, we should inform user about the validation results, as we do for DMN, in the Problems panel.

Image

The existing DMN asset validation has these entry points in the codebase:
- https://github.com/kiegroup/kie-tools/blob/main/packages/online-editor/src/editor/DmnRunner/DmnRunnerService.ts
- https://github.com/kiegroup/kie-tools/blob/main/packages/online-editor/src/editor/DmnRunner/DmnRunnerProvider.tsx

As we try to align problems panel with other editors (DMN and PMML) then for BPMN the problems panel will look like:

Image

### PR DESCRIPTION
**Pull request by jomarko**
This PR closes: https://github.com/kiegroup/kie-issues/issues/36

BPMN assets will be validated via kie-extended-services, what is less or more a wrapper for jitexecutor know from kogito-apps:
- https://github.com/kiegroup/kogito-apps/tree/1.32.x/jitexecutor

Afterwards the validation results will be shown to the developer in the 'Problems' panel at the bottom of the online-editor screen.

### CODE REVIEW

**Comment by yesamer** [comment]
@jomarko You need to update `pnpm-lock.yaml` accordingly with your changes (just run `pnpm bootstrap`)

**Comment by jomarko** [comment]
re https://github.com/kiegroup/kie-tools/pull/1409#issuecomment-1381528754

@yesamer thank you for spotting it, done in separate commit,, my work is still in progress, just wanted to know what CI says to my changes

**Comment by jomarko** [comment]
### TODO
- [x] There are still things named "RUNNER" inside the KieSandboxExtendedServicesClient
- [x] I think the instance of KieSandboxExtendedServicesClient should be provided by the useExtendedServices hook, so components that want to use it, don't have to directly depend on settings to get the URL.
- [x] useKieSandboxExtendedServicesForValidation  -> This hook doesn't need to be coupled with Extended Services on its name, as this is an implementation detail. Components that want to use it won't care how it's done, just that it is done. I think a better name would be useFileValidation , and it could be on a file called "Validation.ts" at src/editor .

File: packages/online-editor/src/editor/DmnRunner/DmnRunnerContext.tsx [hunk 1]
```diff
@@ -29,12 +32,12 @@ export interface DmnRunnerContextType {
   jsonSchema?: DmnSchema;
   mode: DmnRunnerMode;
   didUpdateOutputRows: boolean;
-  service: DmnRunnerService;
+  service: KieSandboxExtendedServicesClient;
```
Comment (tiagobento) [inline packages/online-editor/src/editor/DmnRunner/DmnRunnerContext.tsx]: This doesn't have to be here anymore, as components that want to use `KieSandboxExtendedServicesClient` can do so by calling `useExtendedServices`.

File: packages/online-editor/src/editor/DmnRunner/DmnRunnerProvider.tsx [hunk 2]
```diff
@@ -68,10 +64,7 @@ export function DmnRunnerProvider(props: PropsWithChildren<Props>) {
     return isExpanded ? DmnRunnerStatus.AVAILABLE : DmnRunnerStatus.UNAVAILABLE;
   }, [isExpanded]);
 
-  const service = useMemo(
-    () => new DmnRunnerService(settings.kieSandboxExtendedServices.config.url.jitExecutor),
-    [settings.kieSandboxExtendedServices.config]
-  );
+  const service = extendedServices.client;
```
Comment (tiagobento) [inline packages/online-editor/src/editor/DmnRunner/DmnRunnerProvider.tsx]: Can be inlined.

File: packages/online-editor/src/editor/Validation.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  editorPageDock: EditorPageDockDrawerRef | undefined
+) {
+  const { i18n } = useOnlineI18n();
+  const extendedServices = useExtendedServices();
+  const settings = useSettings();
+
+  const service = useMemo(
+    () => new KieSandboxExtendedServicesClient(settings.kieSandboxExtendedServices.config.url.jitExecutor),
+    [settings.kieSandboxExtendedServices.config]
+  );
```
Comment (tiagobento) [inline packages/online-editor/src/editor/Validation.tsx]: Can be replaced by `extendedServices.client`.

**Review by tiagobento — CHANGES_REQUESTED** [review]
Almost there! Please see my comments inline.

File: packages/online-editor/src/editor/Validation.tsx (hunk 3 above)
Comment (jomarko) [reply packages/online-editor/src/editor/Validation.tsx]: done as d42e23ab50

File: packages/online-editor/src/editor/DmnRunner/DmnRunnerContext.tsx (hunk 1 above)
Comment (jomarko) [reply packages/online-editor/src/editor/DmnRunner/DmnRunnerContext.tsx]: done as be412bc368

File: packages/online-editor/src/editor/DmnRunner/DmnRunnerProvider.tsx (hunk 2 above)
Comment (jomarko) [reply packages/online-editor/src/editor/DmnRunner/DmnRunnerProvider.tsx]: done as be412bc368

**Review by tiagobento — APPROVED** [review]

**Pull request merged**